src/tox/config/sets.py

```diff
@@ -179,6 +179,15 @@ def __init__(self, conf: Config, section: Section, root: Path, src_path: Path) -
         desc = "define environments to automatically run"
         self.add_config(keys=["env_list", "envlist"], of_type=EnvList, default=EnvList([]), desc=desc)
 
+    def _default_work_dir(self, conf: Config, env_name: str | None) -> Path:  # noqa: U100
+        return cast(Path, self["tox_root"] / ".tox")
+
+    def _default_temp_dir(self, conf: Config, env_name: str | None) -> Path:  # noqa: U100
+        return cast(Path, self["work_dir"] / ".tmp")
+
+    def _work_dir_post_process(self, dir: Path) -> Path:
```

Comment:
Why move these to class methods from before being inlined methods?

Reply:
I tend to try and avoid nested functions for a variety of reasons. By defining them as methods it improves testability while simultaneously making `register_config` shorter and easier to digest at first glance.

It's more idiomatic imo -- of course I'm happy to revert if you feel differently!